src/features/entries/slice.js

```diff
@@ -51,50 +166,53 @@ const selectEntryId = (state, props) => {
   }
 }
 
-export const selectAreEntriesLoading = (state) => {
-  return selectIsDatabaseLoading(state, {
-    databaseName,
-  })
-}
+const selectEntriesSlice = (state) => state[SLICE_NAME]
 
-const selectItems = (state) => {
-  return selectDatabaseItems(state, { databaseName })
-}
+export const selectAreEntriesLoading = createSelector(
+  [selectEntriesSlice],
+  (slice) => {
+    return [DB_STATUS.INITIAL, DB_STATUS.OPENING].includes(slice.status)
+  }
+)
 
-const selectPendingItemsById = (state) => {
-  return selectDatabasePendingItemsById(state, { databaseName })
-}
+const selectChanges = createSelector([selectEntriesSlice], (slice) => {
+  return slice.changes
+})
 
-export const selectEntries = createSelector([selectItems], (items) => {
+export const selectEntries = createSelector([selectAll], (items) => {
   return items.map(transformItemToEntry)
 })
 
 export const selectEntriesById = createSelector([selectEntries], (entries) => {
+  // TODO: Use selectEntities, but then the transformation needs to happen in the reducer first
   return keyBy(entries, "entryId")
 })
 
 export const selectEntriesSortedByDate = createSelector(
   [selectEntries],
   (entries) => {
+    // TODO: Replace with selectEntries since they are already sorted by date,
+    // but then the transformation needs to happen in the reducer first
     return sortBy(entries, "date")
   }
 )
 
-export const selectIsEntryPending = createSelector(
-  [selectPendingItemsById, selectEntryId],
-  (pendingByItemId, entryId) => {
-    const pendingItem = first(pendingByItemId[entryId])
-    return !!pendingItem
+export const selectPendingEntry = createSelector(
+  [selectChanges, selectEntryId],
+  (changes, entryId) => {
+    const pendingEntries = changes.filter((change) => {
+      return change.requestStatus === "pending" && change.itemId === entryId
+    })
+    if (pendingEntries.length > 0) {
+      return transformItemToEntry(last(pendingEntries))
+    }
   }
 )
 
-export const selectPendingEntry = createSelector(
-  [selectPendingItemsById, selectEntryId],
-  (pendingByItemId, entryId) => {
-    const pendingItem = first(pendingByItemId[entryId])
-    if (pendingItem) {
-      return transformItemToEntry(pendingItem)
-    }
+export const selectIsEntryPending = createSelector(
```

Comment:
nitpick (non-blocking): Again `selectIsEntryChangePending` would be better

But this might make to many breaking changes in other files and can wait.

Reply:
Not used anywhere so fine...